hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/endpoint/ObjectEndpoint.java

```diff
@@ -1236,6 +1250,10 @@ private CopyObjectResponse copyObject(OzoneVolume volume,
     String sourceBucket = result.getLeft();
     String sourceKey = result.getRight();
     DigestInputStream sourceDigestInputStream = null;
+
+    String sourceBucketOwner = volume.getBucket(sourceBucket).getOwner();
+    String destBucketOwner = volume.getBucket(destBucket).getOwner();
+    S3Owner.verifyBucketOwnerConditionOnCopyOperation(headers, destBucket, sourceBucketOwner, destBucketOwner);
```

Comment:
Since the `destBucket` owner is already checked in the outer scope (`ObjectEndpoint#put`), there is no need to call `getBucket` again. We can simply check the source bucket.